com.unity.ml-agents/Runtime/Sensors/ObservationSpec.cs

```diff
@@ -0,0 +1,90 @@
+using Unity.Barracuda;
+
+namespace Unity.MLAgents.Sensors
+{
+    /// <summary>
+    /// This is the simplest approach, but there's possible user error if Shape.Length != DimensionProperties.Length
+    /// </summary>
+    public struct ObservationSpec
+    {
+        public ObservationType ObservationType;
+        public int[] Shape;
```

Comment:
I wonder if it would be better replace this array with a blittable struct. We know that the maximum length of this Shape is 3 right now.